### ISSUE TITLE
ORM

### ISSUE THREAD

**Issue by Jasperav** [scylladb/scylla-rust-driver#128]
I made an orm for https://github.com/AlexPikalov/cdrs a while ago: https://github.com/Jasperav/cdrs_orm. In short:

- A lot of pre-generated queries (https://github.com/Jasperav/cdrs_orm/blob/master/cdrs_db_mirror/test_derived_equals/src/gen/generated_some_struct.rs)
- Compile time type checked queries: https://github.com/Jasperav/cdrs_orm/blob/master/cdrs_query/cdrs_query_example/src/lib.rs
- Automatic table mapping to Rust structs: https://github.com/Jasperav/cdrs_orm/blob/master/cdrs_to_rust/src/test_result/test_table.rs
- JSON mapping: https://github.com/Jasperav/cdrs_orm/blob/master/cdrs_db_mirror/example_db_mirror/src/lib.rs

I see that Scylla has an ORM for Go: https://github.com/scylladb/gocqlx.

I think it's a good idea to make an ORM for this crate when it is production ready, maybe we can use cdrs_orm (my crate) as a starting point. I can help with the migration.

**Comment by Jasperav** [comment]
I assume it will take a while to make this crate production ready, so for the time being I cancel this issue.

**Comment by psarna** [comment]
It's actually perfectly fine for this issue to stay, even for a long time, so that people know it's on the roadmap - the decision is yours as the author, but feel free to reopen anytime :)

**Comment by Milo123459** [comment]
Any updates? I really want an orm for scylla in Rust

**Comment by Jasperav** [comment]
@Milo123459 I still want to migrate/rewrite the orm, but I won't do it until https://github.com/scylladb/scylla-rust-driver/issues/231 is fixed

**Comment by Milo123459** [comment]
Now that https://github.com/scylladb/scylla-rust-driver/pull/294 is a thing, once that is merged, will this be reopened?

**Comment by Jasperav** [comment]
@Milo123459 Yes once it is merged it takes some time to rewrite the cdrs_orm but I will be working on it :)

**Comment by Milo123459** [comment]
Wooooooooooooo!
